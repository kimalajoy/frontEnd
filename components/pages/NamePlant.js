import React, {Component} from 'react';
import {
  View,
  Text,
  SafeAreaView,
  Input,
  TouchableOpacity,
  Image,
  ImageBackground } from 'react-native';

import styles from '../../styles/styles';
import InputBar from '../InputBar';
import api from '../../api/plantAPI';
import { OpacityButton } from '../buttons/OpacityButton'
import backgroundImg from '../../assets/plant_info_background.jpg';

export default class NamePlant extends Component {
  state = {
    name: ''
  }

  setName = (text) => {
    this.setState({name: text})
  }

  addPlant = () => {
    this.addPlantToGarden(this.props.route.params.id, this.state.name)
    this.props.navigation.dangerouslyGetParent().setOptions({ tabBarVisible: true })
    this.props.navigation.navigate('SearchPage')
  }

  addPlantToGarden = (id, name) => {
    api.addToMyGarden(id, name)
    .then(response => {})
    .catch(err => console.log(err))
  }

  render() {
    const plantImg = this.props.route.params.image;
    return (
      <SafeAreaView style={styles.container}>
        <ImageBackground
          source={backgroundImg}
          style={styles.greenCropBackground}>
        <View style={styles.plantInfoHeader}>
          <Text style={styles.headerText}>Name Your Crop</Text>
        </View>
        <View style={styles.transparentSubHeader}></View>
        <View style={styles.plantImgContainer}>
          <View style={styles.plantImgContainer}>
            <Image
              style={styles.plantImg}
              source={{uri: plantImg}}/>
          </View>
        </View>
        <View style={[styles.addPlantContainer, styles.boxShadow]}>
          <View style={[styles.addNameForm, styles.borderRadius]}>
            <InputBar
              styles={[styles.namePlantInputContainer]}
              placeholder={'Enter plant name'}
              buttonText={'Add Plant'}
              setValue={this.setName}
              inputButtonClick={this.addPlant}/>
          </View>
        </View>
        <OpacityButton
          text={ 'Back to Search' }
          pressFunction={ () => this.props.navigation.goBack() }
          textStyle={ styles.backButton }
          buttonStyle={ styles.backButtonContainer }/>
        </ImageBackground>
      </SafeAreaView>
    )
  }
}
