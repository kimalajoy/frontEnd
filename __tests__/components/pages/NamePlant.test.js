import React from 'react';
import { render, cleanup } from 'react-native-testing-library';

import NamePlant from '../../../components/pages/NamePlant';

describe("Landing Page / Home Screen", () => {
  let namePlant;

  beforeEach(() => {
    namePlant = render(<NamePlant />)
  });

  afterEach(cleanup);

  test('renders "Give your new crop a name" to screen', () => {
    const instruction = namePlant.getByText('Give your new crop a name')

    expect(instruction).toBeTruthy();
  })

  test('has search input bar', () => {
    const inputBar = namePlant.getByPlaceholder('Enter plant name');

    expect(inputBar).toBeTruthy();
  })

  test('has search button', () => {
    const searchButton = namePlant.getByText('Add Plant');

    expect(searchButton).toBeTruthy();
  })

  test('has go back button', () => {
    const goBackButton = namePlant.getByText('Go Back')

    expect(goBackButton).toBeTruthy();
  })

  test('everything renders as expected', () => {
    expect(namePlant.toJSON()).toMatchSnapshot()
  })
})
