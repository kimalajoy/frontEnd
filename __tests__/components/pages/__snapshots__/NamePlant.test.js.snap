// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page / Home Screen everything renders as expected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View>
    <Text>
      Give your new crop a name
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#a7a7a7",
          "padding": 5,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter plant name"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "borderColor": "#1d1d1d",
              "borderRadius": 2,
              "borderWidth": 1,
              "color": "#1d1d1d",
              "height": 30,
              "margin": 10,
              "padding": 5,
              "width": 300,
            },
            Object {
              "color": "#1d1d1d",
              "fontSize": 19,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "borderRadius": 5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Add Plant
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text>
        Go Back
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
